samples/interactive-tutorials/src/main/java/product/AddFulfillmentPlaces.java

```diff
@@ -74,11 +78,12 @@ public static void addFulfillmentPlaces(String productName, String placeId)
     // completing all of your requests, call the "close" method on the client to
     // safely clean up any remaining background resources.
     try (ProductServiceClient serviceClient = ProductServiceClient.create()) {
-      serviceClient.addFulfillmentPlacesAsync(addFulfillmentPlacesRequest);
+      OperationFuture<AddFulfillmentPlacesResponse, AddFulfillmentPlacesMetadata> response =
+              serviceClient.addFulfillmentPlacesAsync(addFulfillmentPlacesRequest);
       // This is a long-running operation and its result is not immediately
       // present with get operations,thus we simulate wait with sleep method.
-      System.out.println("Add fulfillment places, wait 45 seconds: ");
-      TimeUnit.SECONDS.sleep(45);
+      System.out.println("Waiting for operation to finish...");
+      while (!response.isDone()) {}
```

Comment:
Can do either:

AddFulfillmentPlacesResponse response =  serviceClient.addFulfillmentPlacesOperationCallable().call(addFulfillmentPlacesRequest);

or

AddFulfillmentPlacesResponse response = serviceClient.addFulfillmentPlacesAsync(addFulfillmentPlacesRequest).get();

In case the API does not return done, the above methods will time out in 5 minutes and return.

Reply:
This method resulted in exception: error unpacking result. Modified the solution to fix it, now it works without while and exceptions.